### PR TITLE
[#239] Re-export more STM functions.

### LINKED ISSUE #239: Relude.Lifted.Concurrent is awkward
Functions like `retry` and `newTMVar` are not reexported, but if you import `Control.Concurrent.STM` to get them then the lifted definitions of `atomically` etc. conflict with the regular ones.

### PR DESCRIPTION
**Pull request by mithrandi**
Resolves #239

### HLint

- [x] I've changed the exposed interface (add new reexports, remove reexports, rename reexported things, etc.).
  - [x] I've updated [`hlint.dhall`](https://github.com/kowainik/relude/blob/master/hlint/hlint.dhall) accordingly to my changes (add new rules for the new imports, remove old ones, when they are outdated, etc.).
  - [ ] I've generated the new `.hlint.yaml` file (see [this instructions](https://github.com/kowainik/relude#generating-hlintyaml)).

### General

- [ ] I've updated the [CHANGELOG](https://github.com/kowainik/relude/blob/master/CHANGELOG.md) with the short description of my latest changes.
- [x] All new and existing tests pass.
- [x] I keep the code style used in the files I've changed (see [style-guide](https://github.com/kowainik/org/blob/master/style-guide.md#haskell-style-guide) for more details).
- [x] I've used the [`stylish-haskell` file](https://github.com/kowainik/relude/blob/master/.stylish-haskell.yaml).
- [ ] My change requires the documentation updates.
  - [ ] I've updated the documentation accordingly.
- [ ] I've added the `[ci skip]` text to the docs-only related commit's name.

### CODE REVIEW

**Comment by mithrandi** [comment]
Let me know if I should remove names.

File: src/Relude/Lifted/Concurrent.hs [hunk 1]
```diff
@@ -25,17 +25,40 @@ module Relude.Lifted.Concurrent
 
          -- * STM
        , STM
-       , TVar
        , atomically
+       , STM.orElse
+       , STM.check
```
Comment (chshersh) [inline src/Relude/Lifted/Concurrent.hs]: `orElse` and `check` functions also have too generic name. When I see such functions in the code, I don't expect them to work with `STM` immediately, they could do anything and Hoogle search confirms that there're multiple different implementations for those names. In `relude` we don't want to introduce aggressive name conflicts.

Comment (mithrandi) [reply src/Relude/Lifted/Concurrent.hs]: Removed!

**Comment by mithrandi** [comment]
Note to future code spelunkers; `instance Alternative STM where (<|>) = orElse; empty = retry` and so you can use `guard` / `guarded` instead of `check`.

**Pull request merged**